acme/dns_challenge.go

```diff
@@ -87,11 +76,35 @@ func (s *dnsChallenge) Solve(chlng challenge, domain string) error {
 		return err
 	}
 
-	return s.validate(s.jws, domain, chlng.URI, challenge{Resource: "challenge", Type: chlng.Type, Token: chlng.Token, KeyAuthorization: keyAuth})
+	return s.Validate(s.jws, domain, chlng.URI, Challenge{Resource: "challenge", Type: chlng.Type, Token: chlng.Token, KeyAuthorization: keyAuth})
+}
+
+func (s *dnsChallenge) Present(chlng Challenge, domain string) (string, error) {
```

Comment:
If I change `Present` to  `Present(chlng *Challenge, domain string)`, then I can set keyAuth directly in it. Then the return can be just error. Should I do that?